qiskit_experiments/library/characterization/t2ramsey.py

```diff
@@ -103,20 +103,26 @@ def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
             The experiment circuits
 
         Raises:
-            AttributeError: if unit is 'dt', but 'dt' parameter is missing in the backend configuration
+            AttributeError: if unit is 'dt', but 'dt' parameter is missing in the backend configuration.
+            ValueError: if values in `delays` are not strictly increasing.
         """
         if self._unit == "dt":
             try:
                 dt_factor = getattr(backend._configuration, "dt")
             except AttributeError as no_dt:
                 raise AttributeError("Dt parameter is missing in backend configuration") from no_dt
+        if not all(d1 < d2 for d1, d2 in zip(self._delays, self._delays[1:])):
+            raise ValueError("Delays should be strictly increasing")
 
         circuits = []
+        min_delay = self._delays[0] if self._delays[0] != 0 else self._delays[1]
```

Comment:
If we make sure `self._delays` is a numpy array by doing `self._delays = np.asarray(delays)` in the init, then this can be
```suggestion
        min_delay = np.min(self._delays[self._delays > 0])
```